overlord/ifacestate/ifacemgr.go

```diff
@@ -178,6 +179,21 @@ func (m *InterfaceManager) doSetupProfiles(task *state.Task, _ *tomb.Tomb) error
 	}
 	snap.AddImplicitSlots(snapInfo)
 	snapName := snapInfo.Name()
+	var snapState snapstate.SnapState
+	if err := snapstate.Get(task.State(), snapName, &snapState); err != nil {
+		task.Errorf("cannot get state of snap %q: %s", snapName, err)
+		return err
+	}
+
+	// Set DevMode flag if SnapSetup.Flags indicates it should be done
+	// but remember the old value in the task in case we undo.
+	task.Set("old-dev-mode", snapState.DevMode())
+	if ss.Flags&int(snappy.DeveloperMode) != 0 {
```

Comment:
I'm not quite sure about SnapSetup.Flags, it seems there are some tests passing various values there.